src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1054,14 +1055,15 @@ RemoveNodeFromCluster(char *nodeName, int32 nodePort)
 	WorkerNode *workerNode = ModifiableWorkerNode(nodeName, nodePort);
 	if (NodeIsPrimary(workerNode))
 	{
+		if (CanRemoveReferenceTablePlacements())
```

Comment:
I don't want to over  complicate things, but it'd be really nice if the error message is more clear when remove node fails because of reference tables

```C
 if (list_lenght(ReferenceTableOidList()) > 0)
{
if (CanRemoveReferenceTablePlacements())
{
    DeleteAllReferenceTablePlacementsFromNodeGroup(workerNode->groupId);
}
else
{
    ereport (you cannot remove node because there are reference tables and  you would lose the data in reference tables 
     errhint drop all reference  tables before removing the node
}

}
```

And also optionally update the error message for `NodeGroupHasShardPlacements ` check stating that there are placements of  distributed tables

Reply:
>you cannot remove node because there are reference tables and  you would lose the data in reference tables

What if there are reference tables and also distributed tables? Then this error message will first say that there are reference tables and if we drop the reference tables, it will error for distributed tables, however in the first error message it wasn't mentioning about the distributed tables. I guess that might be okay but I am curious about what you think

Reply:
Seems ok to me.

Reply:
Added this block:

https://github.com/citusdata/citus/pull/4127/files#diff-b22dddbebb093ef1571e8253f2edd10aR1069-R1076